### PR TITLE
Compile time schema generation

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

This will move the schema generation to the runtime, as opposed to the existing method of generating the schema in the runtime.

With this change, the GraphQL services will be supported for documentation using the Ballerina documentation.

Fixes: [#2532](https://github.com/ballerina-platform/ballerina-standard-library/issues/2532)
Fixes: [#2698](https://github.com/ballerina-platform/ballerina-standard-library/issues/2698)
Fixes: [#1364](https://github.com/ballerina-platform/ballerina-standard-library/issues/1364)
Fixes: [#2800](https://github.com/ballerina-platform/ballerina-standard-library/issues/2800)

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec

### CODE REVIEW

**Review by shafreenAnfar — APPROVED** [review]
Approving as the changes are verified via the test-cases.

File: ballerina-tests/tests/37_interfaces.bal [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+@test:Config {
+    groups: ["interfaces"]
+}
+isolated function testInterfaceField() returns error? {
+    string document = check getGraphQLDocumentFromFile("interface_field.graphql");
+    string url = "http://localhost:9098/interfaces";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("interface_field.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
```
Comment (DimuthuMadushan) [inline ballerina-tests/tests/37_interfaces.bal:50]: Does the compiler plugin test cover all the negative test cases related to `interfaces`? Don't we need to add negative test cases here?

Comment (ThisaruGuruge) [reply ballerina-tests/tests/37_interfaces.bal:50]: What kind of negative tests are we looking at here? Compiler plugin tests are checking for invalid interface definitions. (We cannot test them in Ballerina tests, right).
